localstack-core/localstack/services/lambda_/event_source_mapping/pollers/stream_poller.py

```diff
@@ -164,7 +170,7 @@ def poll_events_from_shard(self, shard_id: str, shard_iterator: str):
         records = get_records_response.get("Records", [])
         if not records:
             self.shards[shard_id] = get_records_response["NextShardIterator"]
-            raise EmptyPollResultsException(service=self.event_source(), source_arn=self.source_arn)
```

Comment:
that gonna help a lot with responsiveness; good we caught this accidential delay introducer 😬

Reply:
This will effectively prevent backoff when querying empty streams, right?